### ISSUE TITLE
Copy-Item Example 2 and 3 Note is backwards

### ISSUE THREAD

**Issue by LanceMcCarthy** [MicrosoftDocs/PowerShell-Docs#6550]
Using Powershell Core 6

Please review the note in Example 2  https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/copy-item?view=powershell-7#example-2--copy-directory-contents-to-an-existing-directory

My experience with PS Core is that it is the opposite. 

- Without the glob, it adds the source folder to the destination folder.
- When I add the glob, only the contents of the source folder are copied.

If my observations are correct, the documentation is backwards.

### Test 1

This one uses the glob, the source folder is  `\Release\*`

```powershell
Copy-Item -Path "..\bin\Release\*" -Recurse -Destination "..\Binaries" -Force -Container
```

**Result**

According to the documentation, this is supposed to also copy the `Release` folder. However it does not:

![image](https://user-images.githubusercontent.com/3520532/91222493-4061a480-e6ed-11ea-94d0-85e3667065d8.png)

### Test 2

This one does not use a glob, the source folder is just `\Release`

```powershell
Copy-Item -Path "..\bin\Release" -Recurse -Destination "..\Binaries" -Force -Container
```

**Result**

According to the documentation's note, there should not be a `release` folder. However, as you can see here, the folder is present

![image](https://user-images.githubusercontent.com/3520532/91222267-f2e53780-e6ec-11ea-94ba-06b96606510e.png)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: aba5e82d-20c1-2770-1f73-01a8a474d75c
* Version Independent ID: 1eace7de-83aa-39a7-7a19-862e9e37b5f7
* Content: [Copy-Item (Microsoft.PowerShell.Management) - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/copy-item?view=powershell-7#parameters)
* Content Source: [reference/7.0/Microsoft.PowerShell.Management/Copy-Item.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.0/Microsoft.PowerShell.Management/Copy-Item.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @daxian-dbw
* Microsoft Alias: **dongbow**

**Comment by Windos** [comment]
The examples in the docs are working as supplied for me in 5.1, 7.0.3, and 7.1.0-preview.6, I think the behavior being seen here is an interaction between the glob and the `-Container` switch (not part of the examples)

**Comment by Windos** [comment]
Umm @LanceMcCarthy... re-reading the issue here. I think what you're seeing is 100% correct and in line with the docs (at least for example 2).

![image](https://user-images.githubusercontent.com/6955786/91240688-c554c880-e796-11ea-8629-a8808945cffc.png)

According to that, the given example *with* the glob should *not* have the parent source folder, which is what you're seeing.

> ```powershell
> Copy-Item -Path "..\bin\Release\*" -Recurse -Destination "..\Binaries" -Force -Container
> ```
>
> ![image](https://user-images.githubusercontent.com/3520532/91222493-4061a480-e6ed-11ea-94d0-85e3667065d8.png)

This is as promised in the docs, included the glob and the parent directory wasn't included.

> ```powershell
> Copy-Item -Path "..\bin\Release" -Recurse -Destination "..\Binaries" -Force -Container
> ```
>
> ![image](https://user-images.githubusercontent.com/3520532/91222267-f2e53780-e6ec-11ea-94ba-06b96606510e.png)

Again, this is in line with the note on example 2, the parent directory was *included * at the destination when you *removed* the glob.

Example 3 on the other hand is off. This I think needs to be rewarded, especially: `without the subdirectory trees` because the subdirectory trees sure are created in my testing.

**Comment by Windos** [comment]
Just to throw another spanner in the mix (🤦‍♂️🤦‍♂️🤦‍♂️, so much 🤦‍♂️) the behavior is different if the destination folder doesn't exist and has to be created as part of the copy..

**Comment by LanceMcCarthy** [comment]
To help shed some light on how I can across this, here's the line that I was stuck on:

https://github.com/LanceMcCarthy/UI-For-UWP/blob/lance/update-uno/BuildTools/CopyUnoBinaries.ps1#L14

Notice I am using the glob, that produces the opposite result from what Examples 2's note tells me to expect.

![image](https://user-images.githubusercontent.com/3520532/91241955-21c2d400-e714-11ea-97a9-07687aeb7726.png)


Yes! none of the target folders exist at the time I run the command.

**Comment by Windos** [comment]
@LanceMcCarthy, that looks correct to me and in line with the docs?

The inclusion of the glob on line 14 is resulting in *not* getting the parent folder(s) included in the copy jobs.

The note says that if you do want the parent folder then you should remove the glob (and in my testing that works as long as the destination folder exists already.)

**Comment by LanceMcCarthy** [comment]
Okay, I reread the examples again, I believe it is correct. 

It initially seems confusing because the title of the example makes you believe that only the contents are being copied.

To use a concrete example to finally lay this to rest.

1. Source

Source looks like this:

- `..\bin\Release\dotnetstandard20`
- `..\bin\Release\xamarinios12`
- `..\bin\Release\monodroid90`

2. Target

The target is a single empty folder at  `./Binaries`

3. Desired result

I needed everything in `..\bin\Release` to get copied into `.\Binaries` (maintaining original folder structure)

- `.\Binaries\dotnetstandard20`
- `.\Binaries\xamarinios12`
- `.\Binaries\monodroid90`

Attempt 1 (no glob)

When I used `..\bin\release`, I got the `release` folder copied to the target destination

- `./Binaries/Release/dotnetstandard20`
- `./Binaries/Release/xamarinios12`
- `./Binaries/Release/monodroid90`

Attempt 2 (glob)

When I use `bin\release\*`, I get the desired result (the release folder is not in the destination).

- `./Binaries/dotnetstandard20`
- `./Binaries/xamarinios12`
- `./Binaries/monodroid90`

### Conclusion

The documentation is correct.

### Feedback

Adjust the title of the example, as this was the misleading part.

Original:
Example 2: Copy directory contents to an existing directory

Suggested Update:
Example 2: Copy directory and content to an another directory

**Comment by chasewilson** [comment]
Hi @LanceMcCarthy, Thanks for the feedback!

@Windos is correct. The documentation is in line with the behavior. 

The note for example two is as follows: "If you need to include the Logfiles directory in the copy, remove the \* from the Path."
- Meaning, if you only want the contents LogFiles, not the LogFiles folder itself to be copied, keep the `\*` in the **Path** Parameter.

The heading of example two is correct. However, Example three's heading: "Example 3: Copy directory contents to a new directory" could be misleading. This example is shows that an entire directory _and_ it's contents are copied to an existing location. so should read something more like "Example 3: Copy directory and contents to a new directory."

I know it's not exactly your issue with the documentation, but do you think this might make it clearer since the bahavior is correct? 

To summarize:
- `Copy-Item -Path "C:\Test\*" -Destination "C:\OtherTest"` will copy contents of the `Test` folder to the `OtherTest` folder but not the `Test` folder itself.
- `Copy-Item -Path "C:\Test" -Destination "C:\OtherTest"` will copy the `Test` folder and it's contents to the  `OtherTest` Folder

Will Updating the title help?

**Comment by chasewilson** [comment]
Well I'm at least glad we are really close to the same page. Example two is still only copying the contents of the folder not the folder itself.  Example three could be updated to clarify the difference. Is that good with you @LanceMcCarthy?

**Comment by LanceMcCarthy** [comment]
@chasewilson Yep, I think that will help. Thanks!

**Comment by chasewilson** [comment]
@LanceMcCarthy thanks again for the feedback. I've created a Pull Request that's linked above this comment if you want to take a look and leave any more feedback you need there. 

@Windos thanks for all the clarification in this issue as well! 

If either of you have any more feedback, feel free to open a new issue or even check out our [Contributor Guidelines](https://docs.microsoft.com/en-us/powershell/scripting/community/contributing/overview?view=powershell-7) to submit a PR for changes. 

Have a great rest of the day!
